src/model/MultiValueColumn.ts

```diff
@@ -0,0 +1,281 @@
+    this.boxPlotScale
+      .domain([this.min, this.max])
+      .range([0, this.getWidth()])
+
+  }
+
+
+  compare(a: any, b: any, aIndex: number, bIndex: number) {
+    this.sortBy = this.getUserSortBy();
+    const a_val = this.getValue(a, aIndex);
+    const b_val = this.getValue(b, bIndex);
+    const sort: any = new CustomSortCalculation(a_val, b_val);
+    const f = sort[this.sortBy].bind(sort);
+
+    return f();
+  }
+
+
+  getLabel(row: any, index: number) {
+    return '' + this.getValue(row, index);
+  }
+
+  getRaw(row: any, index: number) {
+    return this.accessor(row, index, this.id, this.desc, this.findMyRanker());
+  }
+
+  getValue(row: any, index: number) {
+    var v = this.getRaw(row, index);
+    return (v);
+  }
+
+  getColor(data: any) {
+
+    return this.colorScale(data);
+  }
+
+  calculateCellDimension() {
+
+    return (this.getWidth() / this.bins);
+  }
+
+  getxScale(data) {
+    this.xposScale
+      .domain([0, this.bins - 1])
+      .range([0, this.getWidth()]);
+    return this.xposScale(data);
+  }
+
+  getyScale(data, rowheight) {
+    this.yposScale
+      .domain([this.min, this.max])
+      .range([rowheight, 0]);
+    return this.yposScale(data);
+  }
+
+
+  getbinaryColor() {
+
+    return this.colorrange;
+  }
+
+  getthresholdValue() {
+
+    return this.threshold;
+  }
+
+  getVerticalBarHeight(data, rowheight) {
+
+    if (this.min < this.threshold) {
+      this.verticalBarScale
+        .domain([this.min, this.max])
+        .range([0, rowheight / 2]);
+
+      this.verticalBarHeight = (rowheight / 2 - this.verticalBarScale(data));
+
+    } else {
+      this.verticalBarScale
+        .domain([this.min, this.max])
+        .range([0, rowheight]);
+
+      this.verticalBarHeight = this.verticalBarScale(data);
+    }
+
+    return this.verticalBarHeight;
+
+  }
+
+  getyposVerticalBar(data, rowheight) {
+
+    if (this.min < this.threshold) {
+      this.ypositionVerticalBar = (data < this.threshold) ? (rowheight / 2) : rowheight / 2 - this.getVerticalBarHeight(data, rowheight);   // For positive and negative value
+    } else {
+      this.ypositionVerticalBar = rowheight - this.getVerticalBarHeight(data, rowheight);
+    }
+
+    return this.ypositionVerticalBar;
+  }
+
+  getboxPlotData(data) {
+
+    const minval_arr = Math.min.apply(Math, data);
+    const maxval_arr = Math.max.apply(Math, data);
+
+    data.sort(numSort);
+    const q1 = this.boxPlotScale(d3.quantile(data, 0.25));
+    const median = this.boxPlotScale(d3.quantile(data, 0.50));
+    const q3 = this.boxPlotScale(d3.quantile(data, 0.75));
+    const min_val = this.boxPlotScale(minval_arr);
+    const max_val = this.boxPlotScale(maxval_arr);
+    const boxdata = {min: min_val, median: median, q1: q1, q3: q3, max: max_val};
+
+    return (boxdata);
+
+  }
+
+  getUserSortBy() {
+    return this.sortBy;
+  }
+
+  setUserSortBy(rank) {
+
+    let ascending = false;
+
+    switch (rank) {
+      case 'min':
```

Comment:
I would prefer an enum for the different sorting options to avoid spelling errors